### PR TITLE
FIX-#5098: Stop using append internally.

### LINKED ISSUE #5098: BUG: Deprecation Warning 'series.append' use pandas.concat instead
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd
data = {'id': ["ajajajajajjajajajajajaja", "jajjajajajajajajajajja"]}
df = pd.DataFrame(data)
df['ID_short'] = df['id'].str.slice(0, 3)
```

### Issue Description

The append method is depreciated since pandas 1.4.0 and should not be used anymore. Use pandas.concat instead.
(https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.append.html)
With pandas this warning doesn't occur.

### Expected Behavior

No warnings should appear.

### Error Logs

<details>

```python-traceback

FutureWarning: The series.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.

```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 621bc10e16c21ca4270a4c2a25aa05f814f2686b
python           : 3.9.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.14.0-70.26.1.el9_0.x86_64
Version          : #1 SMP PREEMPT Fri Sep 2 16:07:40 EDT 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

Modin dependencies
------------------
modin            : 0.16.0
ray              : 2.0.0
dask             : 2022.01.1
distributed      : 2022.01.1
hdk              : None

pandas dependencies
-------------------
pandas           : 1.5.0
numpy            : 1.23.2
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 53.0.0
pip              : 21.2.3
Cython           : 0.29.32
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.5
html5lib         : None
pymysql          : 1.0.2
psycopg2         : 2.9.3
jinja2           : 3.1.2
IPython          : 8.4.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2022.7.1
gcsfs            : None
matplotlib       : 3.5.3
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 6.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.0
snappy           : None
sqlalchemy       : 1.4.40
tables           : None
tabulate         : 0.8.10
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5098 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/pandas/dataframe.py [hunk 1]
```diff
@@ -1421,7 +1422,7 @@ def memory_usage(self, index=True, deep=False):  # noqa: PR01, RT01, D200
                 self._query_compiler.memory_usage(index=False, deep=deep)
             )
             index_value = self.index.memory_usage(deep=deep)
-            return Series(index_value, index=["Index"]).append(result)
+            return modin.pandas.concat([Series(index_value, index=["Index"]), result])
```
Comment (vnlitvinov) [inline modin/pandas/dataframe.py:1425]: this feels weird, cannot we instead do `from modin.pandas.general import concat` instead?..

**Review by pyrito — DISMISSED** [review]
Minor comments, but other than that LGTM!

File: modin/pandas/dataframe.py (hunk 1 above)
Comment (mvashishtha) [reply modin/pandas/dataframe.py:1425]: @vnlitvinov `from modin.pandas.general import concat` gives

`ImportError: cannot import name 'DataFrame' from partially initialized module 'modin.pandas.dataframe' (most likely due to a circular import) (/Users/maheshvashishtha/software_sources/modin/modin/pandas/dataframe.py)`

and `from modin.pandas import concat` gives

`ImportError: cannot import name 'concat' from partially initialized module 'modin.pandas' (most likely due to a circular import) (/Users/maheshvashishtha/software_sources/modin/modin/pandas/__init__.py)`

I now have `import modin.pandas as pd` and `pd.concat` instead.

Comment (mvashishtha) [reply modin/pandas/dataframe.py:1425]: Never mind. `import modin.pandas as pd` gives `AttributeError: module 'modin' has no attribute 'pandas'` on python 3.6: https://github.com/modin-project/modin/actions/runs/3202049341/jobs/5230663043

I reverted to `import modin` and calling `modin.pandas.concat`.

**Pull request merged**